### PR TITLE
Tests for flatten, groupby, and aggregators

### PR DESCRIPTION
**Pull request by stardust66**

### CODE REVIEW

File: src/transformers/__tests__/flatten.test.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  });
+
+  test("flatten of empty dataset with one collection", () => {
+    expect(
+      uncheckedFlatten({
+        collections: [{ name: "Cases", labels: {} }],
+        records: [],
+      })
+    );
+  });
```
Comment (P-bibs) [inline src/transformers/__tests__/flatten.test.ts:44]: Is there supposed to be a `.toequal()` here after the expect?

**Review by P-bibs — APPROVED** [review]

**Review by thomascastleman — APPROVED** [review]
Looks good

File: src/transformers/__tests__/flatten.test.ts (hunk 1 above)
Comment (stardust66) [reply src/transformers/__tests__/flatten.test.ts:44]: Yup. Didn't realize.

**Comment by thomascastleman** [comment]
Yeah--should we get this merged anyway though and come back to that when we've got the missing value infrastructure in place? It doesn't seem like we'll have missing values done before we let Bill and Jonathan start reviewing, so maybe we should just try to have all our tests so far merged and we can continue adding later.

**Comment by stardust66** [comment]
Sure. Should I remove the tests that assume a certain behavior for missing values or just leave them in for now?

**Comment by thomascastleman** [comment]
I think they're actually pretty close to what it seems like we're ending up with, so maybe leave them in. We can always take them out / remove later.

**Comment by P-bibs** [comment]
You can always use `test.skip(...)` and leave a `TODO:` to add them back later

**Comment by thomascastleman** [comment]
Can we merge into the main testing branch so it's all in one place?

**Comment by stardust66** [comment]
I'm good with that.

**Pull request merged**